File: first architecture Azure functions/DDI-first-architecture-java/src/main/java/com/function/Injector.java
package com.function;



import java.util.logging.Logger;

import com.azure.cosmos.CosmosClient;
import com.azure.cosmos.CosmosClientBuilder;
import com.azure.cosmos.CosmosContainer;
import com.azure.cosmos.CosmosDatabase;
import com.azure.cosmos.models.CosmosItemRequestOptions;
import com.azure.cosmos.models.PartitionKey;
import com.microsoft.azure.functions.ExecutionContext;


public class Injector {
    private static Injector instance;
    private final CosmosContainer container;
    private Logger logger;
    private final String endpoint = "https://registry.documents.azure.com:443/";
    private final String key = "xegBV4RBjOaCvIUAKLeg1aBwL2kqQGtcJDH5HpMLfAJrgat8l4gnbUVkPg9ouwUv3ZxB7lL98lDIACDbejdGJg==";

    private Injector(ExecutionContext context) {
        CosmosClient client = new CosmosClientBuilder()
                .endpoint(endpoint)
                .key(key)
                .buildClient();

        CosmosDatabase database = client.getDatabase("service-registry");
        this.container = database.getContainer("service-registry4");    
        this.logger = context.getLogger();    
    }

    public static Injector getInstance(ExecutionContext context) {
        if (instance == null) {
            instance = new Injector(context);
            return instance;
        } else {
            instance.logger = context.getLogger();
            return instance;
        }
    }


    private void registerService(String id, String serviceName, String serviceUrl) {
        try {
            Service item = new Service();
            item.setServiceAddress(serviceUrl);
            item.setId(id);
            item.setServiceName(serviceName);
            this.container.createItem(item,new PartitionKey(id), new CosmosItemRequestOptions());
        } catch (Exception e) {
            this.logger.info("Error during service registration: "+ e.getMessage());
        }
    }

    public Service getService(String id) {
        try {
            /* 
            CosmosQueryRequestOptions options = new CosmosQueryRequestOptions();
            //options.setPartitionKey(new PartitionKey(id));
            String query = "SELECT * FROM c WHERE c.id = @id";
            List<SqlParameter> parameters = Collections.singletonList(new SqlParameter("@id", id));
            SqlQuerySpec sqlQuerySpec = new SqlQuerySpec(query, parameters);

            long start = System.currentTimeMillis();
            Iterable<FeedResponse<Service>> result = this.container.queryItems(sqlQuerySpec, options, Service.class).iterableByPage();
            long end = System.currentTimeMillis();

            this.context.getLogger().log(Level.INFO, "Read from DynamoDB table executed in " + (end-start) + " ms");

            for (FeedResponse<Service> item : result) {
                List<Service> items = item.getResults();
                for (Service x : items){
                    if ( x.getId().equals(id)){
                        service = x;
                    }
                    
                }
            }
            */
                long start = System.currentTimeMillis();
                Service service = this.container.readItem(
                                                                                    id,
                                                                                    new PartitionKey(id),
                                                                                    Service.class
                                                                                ).getItem();

                long end = System.currentTimeMillis();
                this.logger.info("Read from DynamoDB table executed in " + (end-start) + " ms");
                //this.context.getLogger().info("response: " +response.toString());

            
                this.logger.info("service: " + service.toString());
                return service;
            
        } catch (Exception e) {
            this.logger.info("Error during service retrieval: "+ e.getMessage());
        }
        return new Service();
        
    }

}

File: first architecture Azure functions/DDI-first-architecture-js4/src/functions/Injector.js
const {CosmosClient}  = require("@azure/cosmos");

const databaseId = "service-registry";
const containerId = "service-registry4";
const client = new CosmosClient(process.env.CosmosDbConnectionSetting)



class Injector {


    constructor() {
        if (!Injector.instance) {
            //this.client = new CosmosClient({ endpoint, auth: { key }});
            //this.client = new CosmosClient(process.env.CosmosDbConnectionSetting)
            //this.database = this.client.database(databaseId);
            //this.container = this.database.container(containerId);
            this.container = client.database(databaseId).container(containerId)
            Injector.instance = this;
        }

        return Injector.instance;
    }


    async registerService(id, serviceName, serviceUrl) {
        try {
            await this.container.items.create({
                id: id,
                ServiceName: serviceName,
                ServiceAddress: serviceUrl
            });
        } catch (error) {
            console.error(`Error during service registration: ${error}`);
        }
    }

    async getService(serviceId) {
        /*try{
            const querySpec = {
                query: "SELECT * FROM c WHERE c.id = @id",
                parameters: [
                    {
                        name: "@id",
                        value: serviceName
                    }
                ]
            };
    
            const { resources: items } = await this.client.items.query(querySpec).fetchAll();
            console.log("item: " + items)
            if (items.length > 0) {
                return items[0];
            } else {
                return null;
            }
            


        } catch (error) {
            console.error(`Error during service retrieval: ${error}`);
        }*/
        
        const start = Date.now()
        const service = (await this.container.item(serviceId, serviceId).read())
        const end = Date.now()
        console.log(`Read from CosmosDB table executed in ${end-start} ms `)
        return {service: service.resource, duration: end-start}
        

    }
}

module.exports = Injector;